app/services/mobile_api_service.rb

```diff
@@ -106,6 +130,13 @@ def validate_response(response)
     fail RequestError.new("Got #{response.status} from #{response.env.url}") unless response.status >= 200 && response.status <= 299
 
     body = JSON.parse(response.body)
+
+    if body["status"] == "fail"
+      data = body["data"]
+
+      fail ValidationError.new("Request returned with validation errors", data["validations"]) if data["errorCode"].present?
```

Comment:
That is how I check if there are validation error, since the code is not the same when on the different apis, I assume that when there is an `errorCode` on the response, that there are validations errors.